dandischema/models.py

```diff
@@ -279,6 +284,18 @@ class BaseType(DandiBaseModel):
     )
     _ldmeta = {"rdfs:subClassOf": ["prov:Entity", "schema:Thing"], "nskey": "dandi"}
 
+    class Config:
+        @staticmethod
+        def schema_extra(schema: Dict[str, Any], model: Type["BaseType"]) -> None:
+            for prop, value in schema.get("properties", {}).items():
+                if prop == "identifier":
+                    options = value.pop("anyOf", None)
+                    if "format" in options[0]:
+                        value.update(**options[0])
+                    else:
+                        value.update(**options[1])
```

Comment:
not sure what `if` is about, and why choosing the first one if has `format` but 2nd one (what if it is not there and just anyOf of 1 item?). Note that `else:` path isn't unit-test covered so YMMV ;-)

Reply:
since the field is a union it should generate two items for anyOf. so this if is really there just in case the order is not deterministic. i have always seen thus far the order to be the same as what is specified in the field definition.